app/common/metrics.py

```diff
@@ -62,7 +62,7 @@ def __init__(self, get_slaves: Callable[[], List['app.master.slave.Slave']]):
     def collect(self) -> Iterator[GaugeMetricFamily]:
         active, idle, dead = 0, 0, 0
         for slave in self._get_slaves():
-            if not slave.is_alive(use_cached=True):
+            if not slave.is_alive(use_cached=True) and not slave.is_shutdown():
```

Comment:
Could you add a comment here, stating that the reason why we have this check here, is that if ```slave.is_shutdown``` is True, then that means we have deliberately, and gracefully removed the slave--and that we don't want to categorize deliberate graceful shutdowns as 'dead slaves'.